### ISSUE TITLE
Strange behaviour after git hook is run (`Error: Cannot find module 'turbo'`)

### ISSUE THREAD

**Issue by nathnhughes** [vercel/turbo#2147]
### What version of Turborepo are you using?

1.5.5

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Mac

### Describe the Bug

We're using [Husky](https://typicode.github.io/husky/#/) to run a pre-commit hook before we commit changes to Github and have come across some strange behaviour.

The hook is just a single command: 

```sh
#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

yarn precommit
```

This runs our pre-commit script in our global package.json: 

```js
{
/* ... */
 "precommit": "turbo run precommit",
/* ... */
}
```

...which in turn runs `lint-staged` in both our ui lib and docs directories to prettify and lint the staged code.

After the `pre-commit.sh` script has run, you are no longer able to run any turbo CLI commands without first running `yarn install`. If you try to, you're presented with the following error:

_(\*local directory omitted for privacy\*)_  

```sh
yarn lint

$ turbo run lint
/<local-directory>/node_modules/turbo/bin/turbo:13
    throw e;
    ^

Error: Cannot find module 'turbo'
Require stack:
- /<local-directory>/node_modules/turbo/node-platform.js
- /<local-directory>/node_modules/turbo/bin/turbo
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.resolve (node:internal/modules/cjs/helpers:108:19)
    at downloadedBinPath (/<local-directory>/node_modules/turbo/node-platform.js:44:44)
    at generateBinPath (/<local-directory>/node_modules/turbo/node-platform.js:67:15)
    at Object.<anonymous> (/<local-directory>/node_modules/turbo/bin/turbo:7:5)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/<local-directory>/node_modules/turbo/node-platform.js',
    '/<local-directory>/node_modules/turbo/bin/turbo'
  ]
}

```

For what it's worth, here's the spec of the machine I'm using: 

```
  Model Name: MacBook Pro
  Model Identifier: MacBookPro18,2
  Chip: Apple M1 Max
  Total Number of Cores: 10 (8 performance and 2 efficiency)
  Memory: 64 GB
  System Firmware Version: 7459.141.1
  OS Loader Version: 7459.141.1
```

### Expected Behavior

Should be able to run `turbo` commands without having to re-install.

### To Reproduce

To reproduce: 

1. clone the following repo: 

```sh
git clone https://github.com/nathnhughes/turborepo-husky-unistall-issue.git
```

2. Make some changes to the files
3. Commit these changes
4. Try to run a turbo command (e.g. `yarn lint`) from the root directory.

**Comment by ThaddeusJiang** [comment]
I had the same issue too.

**Comment by nhducit** [comment]
it might related to https://github.com/vercel/turbo/issues/3317

**Comment by sreekanthjnayar** [comment]
The problem seem to be a case where the TURBO_BINARY_PATH is not being constructed correctly while triggering the command from a shell script like Husky or while invoking the turbo command from a child process. A workaround seems to be to manually set the TURBO_BINARY_PATH in your husky pre-commit script, 
export TURBO_BINARY_PATH=<path_to_turbo_based_on_your_system_architecture>
For example, I use a M processor based Mac and I ended up setting the value to node_modules/turbo-darwin-arm64/bin/turbo (the full path). This is not a final fix, but I was generally curious to see what the issue was.

**Comment by joelybahh** [comment]
> @nathnhughes if you upgrade to turbo@1.7, you can install `turbo` globally, and then your precommit could simply be `turbo precommit`.

This solved it for us, we were having various cross-platform development issues, everyone on windows encountered this bug, and everyone on the latest OS for macbook. By upgrading to latest and installing it globally instead, we now have every platform running turbo, and all husky/git hooks run without this platform-specific error.

**Comment by joelybahh** [comment]
Interestingly, in every case of this error, it only occurred when turbo was called from the git hook (and turbo was local dep), the same command could be ran from the terminal error-free

**Comment by nathnhughes** [comment]
@tknickman @joelybahh can confirm that this did appear to fix the issue. 

Perhaps something worth drawing peoples attention to in the docs if this is to be the workaround going forward.

**Comment by nathanhammond** [comment]
This should be addressed by #5695. If you see this again please open a new issue!
